btk/draw_blends.py

```diff
@@ -257,4 +166,145 @@ def generate(Args, blend_generator, observing_generator, multiprocessing=False,
+        """Draws image of isolated galaxies along with the blend image in the
+        single input band.
+
+        The WLDeblending package (descwl) renders galaxies corresponding to the
+        blend_catalog entries and with observing conditions determined by
+        obs_cond. The rendered objects are stored in the observing conditions
+        class. So as to not overwrite images across different blends, we make a
+        copy of the obs_cond while drawing each galaxy. Images of isolated
+        galaxies are drawn with the WLDeblending and them summed to produce the
+        blend image.
+
+        A column 'not_drawn_{band}' is added to blend_catalog initialized as zero.
+        If a galaxy was not drawn by descwl, then this flag is set to 1.
+
+        Args:
+            blend_catalog: Catalog with entries corresponding to one blend.
+            obs_cond: `descwl.survey.Survey` class describing observing conditions.
+            band(string): Name of band to draw images in.
+
+        Returns:
+            Images of blend and isolated galaxies as `numpy.ndarray`.
+
+        """
+        blend_catalog.add_column(
+            Column(np.zeros(len(blend_catalog)), name="not_drawn_" + band)
+        )
+        galaxy_builder = descwl.model.GalaxyBuilder(
+            obs_cond, no_disk=False, no_bulge=False, no_agn=False, verbose_model=False
+        )
+        pix_stamp_size = np.int(self.stamp_size / self.pixel_scale)
+        iso_image = np.zeros((self.max_number, pix_stamp_size, pix_stamp_size))
+        # define temporary galsim image
+        # this will hold isolated galaxy images that will be summed
+        blend_image_temp = galsim.Image(np.zeros((pix_stamp_size, pix_stamp_size)))
+        mean_sky_level = obs_cond.mean_sky_level
+        for k, entry in enumerate(blend_catalog):
+            iso_obs = copy.deepcopy(obs_cond)
+            try:
+                galaxy = galaxy_builder.from_catalog(
+                    entry, entry["ra"], entry["dec"], band
+                )
+                iso_render = self.draw_isolated(galaxy, iso_obs)
+                iso_image[k] = iso_render.image.array
+                blend_image_temp += iso_render.image
+            except descwl.render.SourceNotVisible:
+                if self.verbose:
+                    print("Source not visible")
+                blend_catalog["not_drawn_" + band][k] = 1
+                continue
+        if self.add_noise:
+            if self.verbose:
+                print("Noise added to blend image")
+            generator = galsim.random.BaseDeviate(seed=np.random.randint(99999999))
+            noise = galsim.PoissonNoise(rng=generator, sky_level=mean_sky_level)
+            blend_image_temp.addNoise(noise)
+        blend_image = blend_image_temp.array
+        return blend_image, iso_image
+
+    def run_mini_batch(self, blend_list, obs_cond):
+        """Returns isolated and blended images for bend catalogs in blend_list
```

Comment:
This function is used for multiprocessing now, is it a problem that it's a class function so we have to share the `self` between different processes?

Reply:
I would say no but it might be the reason why test_multiprocessing gives off an error.

Reply:
Hmm what error where you seeing?

Reply:
An assertion error ; the two arrays are mismatched, but not on all fields ; the first ones give the correct output, leading me to think there is an issue with the multiprocessing specifically.

Reply:
Yeah this is quite strange, thanks! I'll think about this error and try reproducing it